env/classroom_app/apps/serializers.py

```diff
@@ -104,4 +85,51 @@ class AllotedCourseSerializer(serializers.ModelSerializer):
     class Meta:
         model = AllotedCourse
         fields = '__all__'
-        
\ No newline at end of file
+
+
+class AssignmentSerializer(serializers.ModelSerializer):
+    assigned_by = TeacherSerializer()
+    course = CourseSerializer()
+    salary = serializers.SerializerMethodField()
+    def get_salary(self,instance):
+        return instance.assigned_by.salary
+
+    class Meta:
+        model = Assignment
+        fields = '__all__'
+
+    def __init__(self, *args, **kwargs):
+        fields = kwargs.pop('fields', None)
+        super().__init__(*args, **kwargs)
+        if fields is not None:
+            allowed = set(fields)
+            existing = set(self.fields)
+            for field_name in existing - allowed:
+                self.fields.pop(field_name)
+        else:
+            pass
```

Comment:
why is this code required in init method? we can pop fields in the respective save/update method.